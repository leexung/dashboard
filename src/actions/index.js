export const onChange = value => ({ type: 'CHANGE_VALUE', value: value });
