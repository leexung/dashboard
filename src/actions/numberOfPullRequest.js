export const SELECT_CHANNEL = 'SELECT_CHANNEL';
export const REQUEST_POSTS = 'REQUEST_POSTS';
export const RECEIVE_POSTS = 'RECEIVE_POSTS';

export const getChannel = channel => ({
  type: SELECT_CHANNEL,
  channel
});

export const requestPosts = () => ({
  type: REQUEST_POSTS
});
export const receivedPosts = json => {
  return {
    type: 'RECEIVE_POSTS',
    json: json
  };
};

export function fetchPosts() {
  return function(dispatch, getState) {
    dispatch(requestPosts());
    fetch('http://hn.algolia.com/api/v1/search?query=redux')
      .then(response => response.json(), error => console.log('An error occurred.', error))
      .then(json => {
        dispatch(receivedPosts(json));
      });
  };
}
