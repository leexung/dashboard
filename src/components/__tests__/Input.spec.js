import { shallow, mount } from 'enzyme';
import React from 'react';
import ConnectInput, { Input } from '../Input';
import configureStore from 'redux-mock-store';
import { Provider } from 'react-redux';
import { createStore } from 'redux';
import { onChange } from '../../actions/index';
import rootReducer from '../../reducers/index';
import renderer from 'react-test-renderer';

describe('>>>INPUT --- Snapshot', () => {
  let container;

  beforeEach(() => {
    container = shallow(<Input />);
  });

  test('+++ Only 1 input with props value 1 by default', () => {
    expect(container.html()).toMatchSnapshot();
  });
});

describe('>>> INPUT--- REACT-REDUX (Mount + wrapping in <Provider>)', () => {
  const initialState = { value: 10 };
  const mockStore = configureStore();
  let store, wrapper;

  beforeEach(() => {
    store = mockStore(initialState);
    wrapper = mount(
      <Provider store={store}>
        <ConnectInput type='text' />
      </Provider>
    );
  });

  it('+++ render the connected(Input) component', () => {
    expect(wrapper.find(Input).length).toEqual(1);
  });

  it('+++ check Prop matches with initialState', () => {
    expect(wrapper.find(Input).prop('value')).toEqual(initialState.value);
  });

  it('+++ check action on dispatching ', () => {
    let action;
    store.dispatch(onChange(500));
    action = store.getActions();
    expect(action[0].type).toBe('CHANGE_VALUE');
  });
});
describe('>>>H O M E --- REACT-REDUX (actual Store + reducers) more of Integration Testing', () => {
  let store, wrapper;
  const initialState = { value: 10 };

  beforeEach(() => {
    store = createStore(rootReducer);
    wrapper = mount(
      <Provider store={store}>
        <ConnectInput type='text' />
      </Provider>
    );
  });

  it('+++ check Prop matches when  casll dispatch', () => {
    const EXPECT = 500;
    store.dispatch(onChange(EXPECT));
    expect(wrapper.find('input').prop('value')).toBe(500);
    // TODO:
  });

  it('+++ check Prop when change input', () => {
    const input = wrapper.find(Input);
    input.simulate('change', { target: { value: 500 } });
    expect(wrapper.find(Input).prop('value')).toBe(500);
  });
});
