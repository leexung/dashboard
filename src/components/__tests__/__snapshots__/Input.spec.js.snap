// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`>>>INPUT --- Snapshot +++ Only 1 input with props value 1 by default 1`] = `"<input/>"`;
